### DISCUSSION TITLE
Contour plot in the XZ or YZ plane

### DISCUSSION THREAD

**Discussion by jagoosw** [JuliaPlots/Makie.jl#3821] [Q&A] [answered]
Hi all,

I'm not sure this is currently possible, but I'm stuck trying to plot contours in the XZ (or YZ) planes. For example I have a 3D volume:

```
using GLMakie

data = randn(10, 10, 10)

fig = Figure()
ax = Axis3(fig[1, 1])
volume!(ax, [1:10;], [1:10;], [1:10;], -[1:10;], data)
contour!(ax, data[:, :, 10])
```

![test](https://github.com/MakieOrg/Makie.jl/assets/26657828/ac453924-6ef5-4a68-9d5f-73fb12ff34df)

I'd like to plot similar contours on the XZ face but I can't find a way todo that. Since `contour` takes `x::Vector, y::Vector, z::Matrix` I tried just switching which were vectors vs matrix but that doesn't work.

Any suggestion would be great, thanks!

**Comment by ffreyer** [comment, marked as answer]
You can `translate!()` and `rotate!()` a plot to move it around in 3D space, or use `transform!(plot, (plane, offset))` like volumeslices: 
https://github.com/MakieOrg/Makie.jl/blob/714cf7a383a8a87d75ac5875eb42f0e650315274/src/basic_recipes/volumeslices.jl#L40-L52

**Reply by jagoosw** [reply]
Thank you!

**Reply by jagoosw** [reply]
For future reference:
```
using CairoMakie

fig = Figure()

ax = Axis3(fig[1, 1], limits = (1, 10, 1, 10, 1, 10))

data = [i*k for i=1:10, j=1:10,k = 1:10]
colorrange = (1, 100)

top = contour!(ax, [1:10;], [1:10;], data[:, :, 10]; colorrange)

translate!(top, 0, 0, 10)

side = contour!(ax, [1:10;], [1:10;], data[:, end, :]; colorrange)

rotate!(side, √2/2, 0, 0, √2/2)

bottom = contour!(ax, [1:10;], [1:10;], data[:, :, 1]; colorrange)
```
![sides](https://github.com/MakieOrg/Makie.jl/assets/26657828/9aba534a-5ed7-481b-992e-a0d0bf4ccddb)
